components/scream/src/physics/shoc/atmosphere_macrophysics.hpp

```diff
@@ -107,7 +107,33 @@ class SHOCMacrophysics : public scream::AtmosphereProcess
         // TODO: remove *_copy views once SHOC can request a subset of tracers.
         tke_copy(i,k) = tke(i,k);
 
-        qw(i,k) = qv(i,k) + qc(i,k);
+        /*---------------------------------------------------------------------------------
+         *Wet to dry mixing ratios:
+         *-------------------------
+         *Since tracers from the host model (or AD) are  wet mixing ratios and SHOC expects
+         *these tracers in dry mixing ratios, we convert the wet mixing ratios to dry mixing
+         *ratios for all the tracers except TKE (a comment about why leave TKE??).
```

Comment:
comment about TKE: "... except TKE, which is always defined in units of J/(kg of total air) and therefore should be considered to be "wet" when stored in the field manager and also "wet" when used in SHOC.

Reply:
Thanks!